### DISCUSSION TITLE
Resigned RPCS3 savedata is not loadable

### DISCUSSION THREAD

**Discussion by m4st3rP** [bucanero/apollo-ps3#28] [Q&A] [answered]
I copied and resigned a savedata of Metal Gear Solid Peace Walker HD Edition that I have from playing the game in RPCS3 with Apollo Save Tool v1.4.2 but when I try to load it I get the error code 80120003. The savedata works in RPCS3.

These were my exact steps:
Copied the savadata from RPCS3 to the correct folder in a USB drive
Plugged the usb drive into my PS3 and started Apollo
Copied the savedata from my USB drive to the PS3 HDD
Selected "Apply Changes & Resign" on the copied HDD savedata
Started the game and upon loading the savedata getting the error code "80120003"

The savedata:
[NPED00686GAME001.zip](https://github.com/bucanero/apollo-ps3/files/5542924/NPED00686GAME001.zip)

**Comment by bucanero** [comment, marked as answer]
The problem lies in the RPCS3 domain: 
the emulator doesn't create proper PS3 saves because they just store the data without encryption. Most ps3 games encrypt their savegame data, but the emulator has not implemented the encryption/decryption layer at all.
The emulator even miss to create some required PS3 files like `PARAM.PFD`

So before even trying to resign it with Apollo, the data needs to be encrypted back as per the original format and key for that game. A computer tool such as `pfdtool` (linux, mac, win) or Bruteforce Savedata (Windows) could be used to regenerate and encrypt the savegame using a real PS3 save as template. I assume there might be guides online about it.

You can compare with a real PS3 savegame, for example here:
https://metalgearsolidinternationalpage.weebly.com/savegames.html

a real `NPED00686` savegame has a `PARAM.PFD` file, and the `00000000.000` data size is different:
- unencrypted: 314088 bytes
- encrypted: 314096 bytes

**Comment by m4st3rP** [comment]
All right. Sorry for asuming this was an issue with Apollo and thank you for the advice.

**Comment by bucanero** [comment]
> All right. Sorry for asuming this was an issue with Apollo and thank you for the advice.

No worries. I've never tried the emulator, and I had no idea Apollo could actually be used there. 

You can force Apollo to skip the encryption/decryption process for any save-game file, if you edit the `games.conf` file. I haven't tested it, but it might allow you to resign those save files inside the emulator.

You need to edit the `games.conf` file, look for the Title ID of your game, and change the `secure_file_id` line like this:

```
; My game
[BLES12345]
;disc_hash_key=
secure_file_id:UNPROTECTED=554E50524F54454354454447414D4500
```

This change will force Apollo to skip any encrypt/decrypt process, but can't guarantee that it will work in the emulated game. Just test it if you want.

**Reply by GR4J0M3R** [reply]
How can I use Apollo to unlock trophy on RPCS3? Right now I can boot Apollo but trophy list is not shown.

**Reply by bucanero** [reply]
> How can I use Apollo to unlock trophy on RPCS3? Right now I can boot Apollo but trophy list is not shown.

Apollo is meant to be used with a real PS3. The RPCS3 emulator has many differences with a real PS3, including how it stores saves and trophies. Rpcs3 is unsupported and Apollo won't work properly in the emulator.

**Reply by GR4J0M3R** [reply]
> > How can I use Apollo to unlock trophy on RPCS3? Right now I can boot Apollo but trophy list is not shown.
> 
> Apollo is meant to be used with a real PS3. The RPCS3 emulator has many differences with a real PS3, including how it stores saves and trophies. Rpcs3 is unsupported and Apollo won't work properly in the emulator.

Thank you for your answer, I hope it can be applied soon.

**Reply by LordMcMutton** [reply]
Chiming in to say that your solution to force Apollo to skip encryption does, in fact, work- even if the program, itself, says that it doesn't.

It throws the Unable to Reassign error, but the changes take place regardless.

**Reply by bucanero** [reply]
thanks for the confirmation.

I still don't know why RPCS3 won't properly emulate the original save encryption (all the information is already available). 
IMHO, the Emulation community should push in that direction, to get accuracy and compatibility with real PS3 hardware.

**Reply by Masamune3210** [reply]
Probably a combination of just not getting to it yet and not entirely sure how much they want to push the bar, so to speak.

**Reply by JS299990** [reply]
Didn't work for me. I guess it doesn't work for every game. In my case it was Sengoku Basara HD Collection (BLJM60488). I was trying to apply cheats to the save.
